### PR TITLE
hello-world example plugin and service

### PR DESCRIPTION
**Pull request by stephenrlouie**
### Change log

Built an example plugin and provided directions on how to run it and check its status

### TODO

For another PR:
- [x] bundle example
- [x] chain together services as an example

### CODE REVIEW

**Comment by adamschaub** [comment]
If the TODO relates to another PR, make them issues instead.

File: config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# container run docker run -it -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`:/sigma -p 2481:2491 sigmadev/sigma --config /sigma/config.yaml
```
Comment (adamschaub) [inline config.yaml]: why though

File: database/couchbase.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func (cb *Couchbase) insertId(id string, intf interface{}) error {
 	}
 	_, err := cb.BKT.Insert(id, intf, 0)
 	if err != nil {
-		return err
+		return replaceErr(err)
```
Comment (adamschaub) [inline database/couchbase.go]: is this copy pasta from `bundle`?

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#### Optional
+- Running Couchbase container (if you do not have this run with NullDB)
+
+## Hello-World Example
+This example is a simple hello-world service that takes a string input under `Name`. This service will return the output under the fieldName `out` with the string value of `<sigma-operation>: Hello <Name>`. So if I run the service with `Name=Sigma` initially Sigma would attempt to deploy it, the resulting output would read `deploy: Hello Sigma`. See the [example container](https://hub.docker.com/r/sigmadev/hello/)
+
+Optional
+- Run couchbase, we provide a deploy script for a containerized couchbase. `./database/cbdeploy.sh`
+
```
Comment (adamschaub) [inline examples/README.md]: Paths seem to be ambiguous here. `./bin/sigma`, `./examples/hello-world/add-plugin.sh` etc

File: examples/hello-world/hello-service.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "helloSvc",
+  "plugin": "hello/v1",
+  "input": [
```
Comment (meganokeefe) [inline examples/hello-world/hello-service.json]: You will probably need to add `site: local` once the sites stuff lands.

File: database/couchbase.go (hunk 2 above)
Comment (mikeynap) [inline database/couchbase.go]: nah, it's cherry-picked from https://github.com/sigma-dev/sigma/pull/66

Comment (mikeynap) [inline database/couchbase.go]: oh wait... actually idk.

File: examples/hello-world/hello-service.json (hunk 4 above)
Comment (adamschaub) [inline examples/hello-world/hello-service.json]: And by probably you mean definitely

Comment (meganokeefe) [inline examples/hello-world/hello-service.json]: NOT ANYMORE.

File: database/couchbase.go (hunk 2 above)
Comment (stephenrlouie) [inline database/couchbase.go]: it was not. picked it up from master now

**Pull request merged**